db/batch/batch.go

```diff
@@ -0,0 +1,77 @@
+package batch
```

Comment:
why make this a new package? tiny packages are usually not a good idea, as they just make your API harder to use and the code harder to develop. Plus, `db.Batch` is IMO a much nicer name than `batch.Batch` - less repetitive, and it clarifies what its purpose is and where it fits.

Reply:
That makes sense. Another possibility is to write tests in a `db_test` package, so then you can have db_test import badgerdb, and badgerdb importing db - no cycle.